features/fixtures/macos/macOSTestApp/main.m

```diff
@@ -9,5 +9,15 @@
 #import <Cocoa/Cocoa.h>
 
 int main(int argc, const char * argv[]) {
+    [[NSUserDefaults standardUserDefaults] registerDefaults:@{
+        // Disable state restoration to prevent the following dialog being shown after crashes
+        // "The last time you opened macOSTestApp, it unexpectedly quit while reopening windows.
+        //  Do you want to try to reopen its windows again?"
+        // https://developer.apple.com/library/archive/releasenotes/AppKit/RN-AppKitOlderNotes/index.html#10_7StateRestoration
+        @"ApplePersistenceIgnoreState": @YES,
+        // Stop NSApplication swallowing NSExceptions thrown on the main thread.
+        @"NSApplicationCrashOnExceptions": @YES,
```

Comment:
This should align with the [docs](https://docs.bugsnag.com/platforms/macos/#reporting-unhandled-exceptions), with either the tests and the docs advocating for this defaults change or by subclassing NSApplication

Reply:
Subclassing NSApplication like the docs suggest will not result in the app crashing as expected by our test scenarios, and so the test will not pass.

Reply:
Right, no argument there. I'm saying that if that's what's expected in the tests, then the docs should match. Recommending that people turn it on when configuring the library (or more generally to avoid corrupting app state) also seems like a reasonable position.

Reply:
https://github.com/bugsnag/docs.bugsnag.com/pull/1256